lib/prefix.h

```diff
@@ -307,6 +307,13 @@ struct prefix_sg {
 	struct in_addr grp;
 };
 
+struct prefix6_sg {
```

Comment:
I have already commented on #10192 that this is not a good way to approach this. (cf. https://github.com/FRRouting/frr/pull/10192#pullrequestreview-827997733 )

I would like to hear what particular reasons led you to ignore these comments.

Reply:
Hi David,
I have not ignored the comments mentioned (cf. #10192 (review) )
I didnt remove prefix_sg from lib since it is used by tests/lib/test_printfrr.c
As per your comment I have introduced pim_prefix_sg separtely for pimd, which will be used for both IPV4 and IPV6 in future.